src/AzureExtension/Widgets/AzurePullRequestsDeveloperWidget.cs

```diff
@@ -0,0 +1,193 @@
+
+namespace DevHomeAzureExtension.Widgets;
+
+internal sealed class AzurePullRequestsDeveloperWidget : AzurePullRequestsBaseWidget
+{
+    private readonly CacheManager _cacheManager;
+
+    public AzurePullRequestsDeveloperWidget()
+    : base()
+    {
+        SupportsCustomization = false;
+        DeveloperIdLoginRequired = false;
+        _cacheManager = CacheManager.GetInstance();
+        LoadingMessage = Resources.GetResource("Widget_Template/LoadingFindingPullRequests", Log);
+    }
+
+    // Data loading methods
+    public override void HandleDataManagerUpdate(object? source, DataManagerUpdateEventArgs e)
+    {
+        if (e.Kind == DataManagerUpdateKind.Developer)
+        {
+            Log.Debug($"Received developer pull request update.");
+
+            if (e.Kind == DataManagerUpdateKind.Error)
+            {
+                DataState = WidgetDataState.FailedUpdate;
+                DataErrorMessage = e.Context.ErrorMessage;
+
+                // The DataManager log will have detailed exception info, use the short message.
+                Log.Error($"Developer pull request update failed. {e.Context.ErrorMessage}");
+
+                if (!LoadedDataSuccessfully)
+                {
+                    SetLoading();
+                }
+
+                return;
+            }
+
+            DataState = WidgetDataState.Okay;
+            DataErrorMessage = string.Empty;
+            LoadContentData();
+        }
+
+        // If we failed data state, any data update might be an opportunity to retry since any
+        // update means a transaction has completed.
+        if (DataState == WidgetDataState.FailedRead)
+        {
+            Log.Debug("Retrying datastore read.");
+            LoadContentData();
+            return;
+        }
+    }
+
+    protected override bool ValidateConfiguration(WidgetActionInvokedArgs args)
+    {
+        return true;
+    }
+
+    public override void RequestContentData()
+    {
+        // This widget does not request data, so we will use the periodic update
+        // as a refresh to try to recover from any transient issue where
+        // data was unavailable for whatever reason.
+        LoadContentData();
+    }
+
+    protected override void ResetDataFromState(string data)
+    {
+        // This widget has no state.
+        return;
+    }
+
+    public override string GetConfiguration(string data)
+    {
+        // This widget has no configuration.
+        return string.Empty;
+    }
+
+    public override void LoadContentData()
+    {
+        if (_cacheManager.NeverUpdated)
+        {
+            SetLoading();
+            return;
+        }
+
+        try
+        {
+            // This can throw if DataStore is not connected.
+            var pullRequestsList = DataManager!.GetPullRequestsForLoggedInDeveloperIds();
+            var itemsData = new JsonObject();
+            var itemsArray = new JsonArray();
+
+            foreach (var pullRequests in pullRequestsList)
```

Comment:
NIT: pullRequestList.Where(x => x is not null)

Reply:
NIT: pullRequestList.Where(x => x is not null).Select(x => JsonConvert.DeserializeObject<Dictionary<string, object>>(x.Results))

Reply:
Took first suggestion, that simplified the code and was clear. Did not do the second, as that started to get a bit too complicated for readability and made for a very long line.